src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -336,4 +502,321 @@ private void populateQSYSMonitoringMetrics(Map<String, String> stats) {
 			throw new ResourceNotReachableException(errBuilder.toString());
 		}
 	}
+
+	/**
+	 * This method is used for populate Q-SYS gain controlling metrics
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param controllableProperties is the list that store all controllable properties
+	 */
+	private void populateGainControllingMetrics(Map<String, String> stats, List<AdvancedControllableProperty> controllableProperties) {
+		// If user doesn't input anything, just break out the method
+		if (gain == null) {
+			return;
+		}
+
+		String[] namedGainComponents = handleGainInputFromUser();
+		GetComponentsResponse namedComponents = (GetComponentsResponse) getResponsesFromDevice(new GetComponentsRequest(), GetComponentsResponse.class);
+
+		// If there is no component in the design file -> just return and do nothing
+		if (namedComponents == null) {
+			return;
+		}
+
+		for (String namedGainComponent : namedGainComponents) {
+			GainControlInfo controlInfo;
+			String errorMessage = null;
+
+			try {
+				controlInfo = getControlInfoFromComponent(namedComponents, namedGainComponent);
+
+				if (controlInfo == null) {
+					errorMessage = QSYSCoreConstant.GETTING_DEVICE_INFO_ERR;
+
+					if (this.logger.isDebugEnabled()) {
+						this.logger.debug(errorMessage);
+					}
+
+					populateUnavailableGainControllingGroup(stats, namedGainComponent, errorMessage);
+				} else {
+					double minGain = Double.parseDouble(controlInfo.getMinGain());
+					double maxGain = Double.parseDouble(controlInfo.getMaxGain());
+
+					// swap 2 value then set back to control info
```

Comment:
could you please add more info to the inline comment here, on why this swap action is needed exactly?

Reply:
sure, I will do it